### PR TITLE
Add bridge for instance cache and reset

### PR DESCRIPTION
**Pull request by ovstinga**

### CODE REVIEW

**Review by andrei-marinica — APPROVED** [review]

File: wasmer2/wasmer2Executor.go [hunk 1]
```diff
@@ -112,8 +106,29 @@ func (wasmerExecutor *Wasmer2Executor) NewInstanceFromCompiledCodeWithOptions(
 	compiledCode []byte,
 	options executor.CompilationOptions,
 ) (executor.Instance, error) {
-	// return NewInstanceFromCompiledCodeWithOptions(compiledCode, options)
-	return nil, errors.New("NewInstanceFromCompiledCodeWithOptions not implemented")
+	var c_instance *cWasmerInstanceT
+
+	if len(compiledCode) == 0 {
+		var emptyInstance = &Wasmer2Instance{cgoInstance: nil}
+		return emptyInstance, newWrappedError(ErrInvalidBytecode)
```
Comment (camilbancioiu) [inline wasmer2/wasmer2Executor.go:113]: Return `nil` here instead of `emptyInstance`.

File: wasmer2/wasmer2Executor.go [hunk 2]
```diff
@@ -112,8 +106,29 @@ func (wasmerExecutor *Wasmer2Executor) NewInstanceFromCompiledCodeWithOptions(
...
+		wasmerExecutor.cgoExecutor,
+		&c_instance,
+		(*cUchar)(unsafe.Pointer(&compiledCode[0])),
+		cUint32T(len(compiledCode)),
+		(*cWasmerCompilationOptions)(cOptions),
+	)
+
+	if compileResult != cWasmerOk {
+		var emptyInstance = &Wasmer2Instance{cgoInstance: nil}
+		return emptyInstance, newWrappedError(ErrFailedInstantiation)
```
Comment (camilbancioiu) [inline wasmer2/wasmer2Executor.go:127]: Same as above, `nil` instead of `emptyInstance`.

File: wasmer2/wasmer2Executor.go [hunk 3]
```diff
@@ -112,8 +106,29 @@ func (wasmerExecutor *Wasmer2Executor) NewInstanceFromCompiledCodeWithOptions(
...
+	}
+
+	instance, err := newInstance(c_instance)
+	return instance, err
```
Comment (camilbancioiu) [inline wasmer2/wasmer2Executor.go:131]: `return newInstance(c_instance)`

File: wasmer2/wasmer2Instance.go [hunk 4]
```diff
@@ -124,27 +124,24 @@ func (instance *Wasmer2Instance) GetBreakpointValue() uint64 {
...
+		return nil, ErrCachingFailed
+	}
 
-	// C.free(unsafe.Pointer(cacheBytes))
-	// cacheBytes = nil
-	// return goBytes, nil
+	goBytes := C.GoBytes(unsafe.Pointer(cacheBytes), C.int(cacheLen))
 
-	return nil, ErrCachingFailed
+	C.free(unsafe.Pointer(cacheBytes))
```
Comment (camilbancioiu) [inline wasmer2/wasmer2Instance.go:142]: :+1:

File: wasmer2/wasmer2Executor.go [hunk 5]
```diff
@@ -98,11 +97,6 @@ func (wasmerExecutor *Wasmer2Executor) NewInstanceWithOptions(
 	}
 
 	instance, err := newInstance(c_instance)
-	// if instance != nil && instance.Memory != nil {
-	// 	c_instance_context := cWasmerInstanceContextGet(c_instance)
-	// 	instance.InstanceCtx = IntoInstanceContextDirect(c_instance_context)
-	// }
-
 	return instance, err
```
[the remark this answers is not in the document]
Comment (ovstinga) [reply wasmer2/wasmer2Executor.go:100]: done

File: wasmer2/bridge2.go [hunk 6]
```diff
@@ -18,21 +18,16 @@ type cUint C.uint
...
-
-// type cWasmerMemoryT C.vm_exec_memory_t
 type cWasmerResultT C.vm_exec_result_t
-
 type cWasmerCompilationOptions C.vm_exec_compilation_options_t
 type cWasmerVmHookPointers = C.vm_exec_vm_hook_c_func_pointers
 
+// type cWasmerByteArray C.vm_exec_byte_array
+// type cWasmerInstanceContextT C.vm_exec_instance_context_t
+// type cWasmerMemoryT C.vm_exec_memory_t
```
[the remark this answers is not in the document]
Comment (ovstinga) [reply wasmer2/bridge2.go:30]: yes, deleted all commented code from this file

File: wasmer2/wasmer2Executor.go (hunk 1 above)
Comment (ovstinga) [reply wasmer2/wasmer2Executor.go:113]: done

File: wasmer2/wasmer2Executor.go (hunk 2 above)
Comment (ovstinga) [reply wasmer2/wasmer2Executor.go:127]: done

**Review by andrei-marinica — APPROVED** [review]

**Pull request merged**